components/Events/Filters.js

```diff
@@ -0,0 +1,228 @@
+        const date = new Date(item.start);
+        const itemDay = Object.keys(chosenDays)[date.getDay()];
+        return chosenDays[itemDay];
+      });
+    }
+    // Check location filter
+    const isLocEmpty = Object.keys(chosenLoc).every((loc) => !chosenLoc[loc]);
+    if (!isLocEmpty) {
+      allEvents = allEvents.filter((item) => {
+        const itemLoc = item.location.toLowerCase().includes('zoom') ? 'Online' : 'In-Person';
+        return chosenLoc[itemLoc];
+      });
+    }
+
+    const newEvents = allEvents.map((original_event, index) => ({
+      ...original_event,
+      id: index,
+    }));
+    props.handleChange(newEvents);
+  };
+
+  const handleWeekdayChange = (day) => {
+    const updated = { ...chosenDays };
+    updated[day] = !updated[day];
+    setChosenDays(updated);
+  };
+
+  const handleCommChange = (comm) => {
+    const updated = { ...chosenComms };
+    updated[comm] = !updated[comm];
+    setChosenComms(updated);
+  };
+
+  const handleTimeChange = (key, value) => {
+    const updated = { ...chosenTimes };
+    updated[key] = value;
+    setChosenTimes(updated);
+  };
+
+  const handleLocationChange = (loc) => {
+    const updated = { ...chosenLoc };
+    updated[loc] = !updated[loc];
+    setChosenLoc(updated);
+  };
+
+  return (
+    <div className={styles.filter}>
+      <div className={styles.inner}>
+        <input
+          placeholder="Search Events"
+          className={styles.searchbar}
+          value={searchValue}
+          onChange={(e) => setSearchValue(e.target.value)}
+          onKeyPress={(e) => e.key === 'Enter' && handleSearch()}
+        />
+        <button onClick={handleSearch} className={styles.searchbutton}>
+          search
+        </button>
+        <button
+          onClick={() => setIsFilterOpen(!isFilterOpen)}
+          className={styles.searchbutton}
+        >
+          filter
+          <FontAwesomeIcon
+            className={styles.arrow}
+            icon={isFilterOpen ? faChevronUp : faChevronDown}
+          />
+        </button>
+        {isFilterOpen && (
+          <div className={styles.filteroptions}>
+            <div>
+              <h3 className={styles.header}>Committees </h3>
+              {Object.keys(chosenComms).map((comm) => (
+                <div key={comm}>
+                  <input
+                    type="checkbox"
+                    id={comm}
+                    checked={chosenComms[comm]}
+                    onChange={() => handleCommChange(comm)}
+                  />{' '}
+                  {comm}
+                </div>
+              ))}
+            </div>
+            <div>
+              <h3 className={styles.header}>Days</h3>
+              {Object.keys(chosenDays).map((day) => (
+                <div key={day}>
+                  <input
+                    type="checkbox"
+                    id={day}
+                    checked={chosenDays[day]}
+                    onChange={() => handleWeekdayChange(day)}
```

Comment:
There can potentially be a mismatch, so maybe taking the direct `checked` value would be more consistent! Same with the other checkboxes

```suggestion
                    onChange={(e) => handleWeekdayChange(day, e.target.checked)}
```